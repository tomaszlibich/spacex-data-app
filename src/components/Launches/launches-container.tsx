import { useEffect, useState, useRef } from "react";
import { getLaunches } from "../../utils/fetch";
import { LaunchesContext, initialData } from "./context";
import { LaunchesMain } from "./launches-main";
import { Launch, LaunchesSortingTypes } from "./types";

export const LaunchesContainer = () => {
  const fetchedLaunchesRef = useRef<Launch[] | null>(null);
  const [launches, setLaunches] = useState(initialData.launches);
  const [sorting, setSorting] = useState<LaunchesSortingTypes | null>(null);
  const [search, setSearch] = useState<string | undefined>();
  const context = {
    launches,
    setSorting,
    sorting,
    search,
    setSearch,
  };

  useEffect(() => {
    if (fetchedLaunchesRef.current === null) {
      (async function fetchData() {
        try {
          const fetchedLaunches = await getLaunches();

          fetchedLaunchesRef.current = fetchedLaunches;
          console.warn("LAUNCHES", fetchedLaunches);
          setLaunches(fetchedLaunches);
        } catch (e) {
          console.error(e);
        }
      })();
    }
  }, []);

  return (
    <LaunchesContext.Provider value={context}>
      <LaunchesMain />
    </LaunchesContext.Provider>
  );
};
