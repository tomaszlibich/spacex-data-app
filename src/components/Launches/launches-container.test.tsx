import renderer from 'react-test-renderer';
import { LaunchesContainer } from "./launches-container";

let component;
let tree: any;

describe("LaunchesContainer", () => {
  describe('When component is rendered', () => {
    beforeEach(() => {
      component = renderer.create(
        <LaunchesContainer />,
      );
      tree = component.toJSON();
    });

    it("should match the snapshot", () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
