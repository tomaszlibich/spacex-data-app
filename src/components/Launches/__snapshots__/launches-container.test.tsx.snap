// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchesContainer When component is rendered should match the snapshot 1`] = `
Array [
  <h1
    className="listHeader"
  >
    Most recent SpaceX rocket launches
  </h1>,
  <div
    className="actions"
  >
    <div
      className="sorting"
    >
      <button
        className=""
        onClick={[Function]}
      >
        Sort by name
      </button>
      <button
        className=""
        onClick={[Function]}
      >
        Sort by date
      </button>
    </div>
    <div
      className="search"
    >
      <input
        name="search"
        onChange={[Function]}
        placeholder="Search..."
        type="text"
      />
    </div>
  </div>,
  <div
    className="list"
  >
    <p
      className="noData"
    >
      No launch data available...
    </p>
  </div>,
]
`;
